sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/NetworkInterfaceImpl.java

```diff
@@ -60,11 +65,16 @@ class NetworkInterfaceImpl
     private NetworkSecurityGroup existingNetworkSecurityGroupToAssociate;
     /** cached related resources. */
     private NetworkSecurityGroup networkSecurityGroup;
+    /** the name of specified ip config name */
+    private Map<DeleteOptions, HashSet<String>> specifiedIpConfigNames;
+    private boolean isInUpdateIpConfigMode = false;
```

Comment:
Will `specifiedIpConfigNames` get cleared after each create/update? And what's the purpose of the extra `isInUpdateIpConfigMode`?

Reply:
1. The `specifiedIpConfigNames` was initialized at [NetworkInterfaceImpl.java#L77](https://github.com/v-hongli1/azure-sdk-for-java/blob/issue%2338806_nic/sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/NetworkInterfaceImpl.java#L77) and [NetworkInterfaceImpl.java#L596](https://github.com/v-hongli1/azure-sdk-for-java/blob/issue%2338806_nic/sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/NetworkInterfaceImpl.java#L596)
2. Update `DeleteOption` of  `PublicIPAddress` when `isInUpdateIpConfigMode = true` only,  it was initialized at [NetworkInterfaceImpl.java#L76](https://github.com/v-hongli1/azure-sdk-for-java/blob/issue%2338806_nic/sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/NetworkInterfaceImpl.java#L76) and [NetworkInterfaceImpl.java#L597](https://github.com/v-hongli1/azure-sdk-for-java/blob/issue%2338806_nic/sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/NetworkInterfaceImpl.java#L597)